src/laddie/laddie_main.f90

```diff
@@ -25,41 +32,83 @@ MODULE laddie_main
 
     ! Local variables:
     CHARACTER(LEN=256), PARAMETER                         :: routine_name = 'run_laddie_model'
-    INTEGER                                               :: vi
-    REAL(dp), PARAMETER                                   :: T_off  = 0.0_dp  ! [degC] Initial temperature offset
-    REAL(dp), PARAMETER                                   :: S_off  = -0.1_dp ! [PSU]  Initial salinity offset
+    INTEGER                                               :: vi, ti
+    REAL(dp)                                              :: tl               ! [s] Laddie time
+    REAL(dp)                                              :: dt               ! [s] Laddie time step
  
     ! Add routine to path
     CALL init_routine( routine_name)
- 
-    ! Initialise ambient T and S
-    CALL compute_ambient_TS( mesh, BMB%laddie, ocean, ice)
 
-    ! Initialise main T and S
+    ! == Preparation ==
+    ! =================
+
+    ! Get time step
+    tl = 0.0_dp
+    dt = C%dt_laddie
+
+    ! == Update masks ==
+    CALL update_laddie_masks( mesh, ice, laddie)
+
+    ! Extrapolate new cells
+    ! TODO, use Gaussian extrap routine
+
+    ! Set values to zero if outside laddie mask
     DO vi = mesh%vi1, mesh%vi2
-       IF (ice%mask_floating_ice( vi)) THEN
-         BMB%laddie%T( vi)      = BMB%laddie%T_amb( vi) + T_off
-         BMB%laddie%T_prev( vi) = BMB%laddie%T_amb( vi) + T_off
-         BMB%laddie%T_next( vi) = BMB%laddie%T_amb( vi) + T_off
-         BMB%laddie%S( vi)      = BMB%laddie%S_amb( vi) + S_off
-         BMB%laddie%S_prev( vi) = BMB%laddie%S_amb( vi) + S_off
-         BMB%laddie%S_next( vi) = BMB%laddie%S_amb( vi) + S_off
-       END IF
+      IF (.NOT. laddie%mask_a( vi)) THEN
+        laddie%now%H( vi)     = 0.0_dp
+        laddie%now%T( vi)     = 0.0_dp
+        laddie%now%S( vi)     = 0.0_dp
+        laddie%melt( vi)  = 0.0_dp
+        laddie%entr( vi)  = 0.0_dp
+      END IF
     END DO
- 
+
+    DO ti = mesh%ti1, mesh%ti2
+      IF (.NOT. laddie%mask_b( ti)) THEN
+        laddie%now%U( ti)     = 0.0_dp
+        laddie%now%V( ti)     = 0.0_dp
+        laddie%now%H_b( ti)   = 0.0_dp
+      END IF
+    END DO
+
+    laddie%H_c = 0.0_dp
+
+    ! Update ice shelf draft gradients
+    CALL ddx_a_b_2D( mesh, ice%Hib , laddie%dHib_dx_b)
+    CALL ddy_a_b_2D( mesh, ice%Hib , laddie%dHib_dy_b)
+
+    ! == Main time loop ==
+    ! ====================
+
+    DO WHILE (tl < C%time_duration_laddie * sec_per_day)
+
+      SELECT CASE(C%choice_laddie_integration_scheme)
+        CASE DEFAULT
+          CALL crash('unknown choice_laddie_integration_scheme "' // TRIM( C%choice_laddie_integration_scheme) // '"')
+        CASE ('euler')
+          CALL integrate_euler( mesh, ice, ocean, laddie, tl, dt)  
+        CASE ('fbrk3')
+          CALL integrate_fbrk3( mesh, ice, ocean, laddie, tl, dt)  
+      END SELECT
+
+      ! Display or save fields
+      CALL print_diagnostics( laddie, tl)
```

Comment:
Is this still needed?

Reply:
Yes, I'll leave this now until I have implemented output of laddie fields/scalars at given intervals in 'laddie time'. I'll make an issue for that.